dev-client/src/components/dataRequirements/handleMissingData.ts

```diff
@@ -44,3 +52,30 @@ export const usePopNavigationAndShowSyncError = () => {
     }
   }, [navigation, syncNotifications]);
 };
+
+export const useNavToSiteAndShowSyncError = (siteId: string) => {
+  const navigation = useNavigation();
+  const syncNotifications = useSyncNotificationContext();
+
+  return useCallback(() => {
+    navigation.navigate('SITE_TABS', {siteId: siteId});
+    if (isFlagEnabled('FF_offline')) {
+      syncNotifications.showError();
+    }
+  }, [siteId, navigation, syncNotifications]);
+};
+
+export const useDefaultSiteDepthRequirements = (
+  siteId: string,
+  depthIntervalSpec: DepthInterval,
+) => {
+  const site = useSelector(selectSite(siteId));
+  const realDepthInterval = useSiteSoilInterval(siteId, depthIntervalSpec);
+  const handleMissingSite = useNavToBottomTabsAndShowSyncError();
+  const handleMissingDepth = useNavToSiteAndShowSyncError(siteId);
+  const requirements = useMemoizedRequirements([
+    {data: site, doIfMissing: handleMissingSite},
+    {data: realDepthInterval, doIfMissing: handleMissingDepth},
+  ]);
+  return requirements;
+};
```

Comment:
nit: i'm not sure if this exactly fits in `handleMissingData.ts`. maybe there should be another file specifically for factoring out common requirement lists like this?